src/main/java/uk/gov/hmcts/ccd/domain/service/DeleteUserProfileJurisdictionOperation.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.ccd.domain.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.userprofile.UserProfileRepository;
+import uk.gov.hmcts.ccd.domain.model.Jurisdiction;
+import uk.gov.hmcts.ccd.domain.model.UserProfile;
+import uk.gov.hmcts.ccd.endpoint.exception.BadRequestException;
+
+@Service
+public class DeleteUserProfileJurisdictionOperation {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DeleteUserProfileJurisdictionOperation.class);
+    private static final String WORKBASKET_DEFAULTS_ERROR = "Cannot delete user profile as the user's workbasket "
+        + "defaults are set to the Jurisdiction the user is being deleted from. Please update the user's workbasket "
+        + "default values to another Jurisdiction and try again.";
+    private final UserProfileRepository userProfileRepository;
+
+    @Autowired
+    public DeleteUserProfileJurisdictionOperation(final UserProfileRepository userProfileRepository) {
+        this.userProfileRepository = userProfileRepository;
+    }
+
+    public UserProfile deleteAssociation(final String userId, final String jurisdictionId) throws BadRequestException {
+        final UserProfile userProfile = userProfileRepository.findById(userId);
+
```

Comment:
L27..L31 can be shortened to final UserProfile userProfile = Optional.ofNullable(userProfileRepository.findById(userId)).orElseThrows(new BadRequestException("User does not exisit with ID" + userId);

Reply:
Good spot, though you missed one bit: it should be
```.orElseThrow(() -> new BadRequestException(...));```